File: src/my_py_pkg/my_py_pkg/node_template.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node


class MyCustomNode(Node):
    def __init__(self, nodeName):
        self.nodeName = nodeName
        super().__init__(nodeName)
        self.get_logger().info(f"Node {nodeName} started.")


def main(args=None):
    rclpy.init(args=args)
    node = MyCustomNode("node_name")
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/node_template.cpp
#include "rclcpp/rclcpp.hpp"

class MyCustomNode : public rclcpp::Node
{
public:
    MyCustomNode(std::string nodeName) : Node(nodeName)
    {
        _nodeName = nodeName;
        RCLCPP_INFO(this->get_logger(), (std::string("ROS2 ") + nodeName + std::string(" C++ started.")).c_str());
    }

private:
    std::string _nodeName;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyCustomNode>(std::string("node_name"));
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}